metrics/tagindex.go

```diff
@@ -0,0 +1,132 @@
+package metrics
+
+import (
+	"sync"
+
+	"github.com/mstoykov/atlas"
+)
+
+// FIXME: naming! we now have TagMap, SampleTags, SystemTag, TagSet
+// so the naming decision and maybe some sort of uniquification could be good.
+//
+// TagSet should be the best but is already used, TagIndex could be a valid alternative?
+
+// TagMap represents a set of tags.
+type TagMap struct {
+	mutex sync.RWMutex
+	tags  *atlas.Node
+	locks bool
+}
+
+// RootTagMap initializes a tag map with a root Atlas node.
+// It should be only used in a centralized place and all the new
+// TagMap branched form it.
+func RootTagMap() *TagMap {
+	return &TagMap{
+		tags:  atlas.New(),
+		mutex: sync.RWMutex{},
+		locks: false,
+	}
+}
+
+// NewTagMap creates a TagMap,
+// if a not-nil map is passed then it will be used as the internal map
+// otherwise a new one will be created.
+//
+// TODO: this should be removed from the code asap
```

Comment:
Mark is [deprecated](https://rakyll.org/deprecated/)

Reply:
This is same as https://github.com/grafana/k6/pull/2594#discussion_r936304976 and it will be fixed with https://github.com/grafana/k6/pull/2633